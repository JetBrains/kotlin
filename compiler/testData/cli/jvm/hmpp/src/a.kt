expect class A {
    fun foo()
}

expect class B {
    fun bar()
}
