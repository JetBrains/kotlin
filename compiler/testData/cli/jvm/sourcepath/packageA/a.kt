package packageA
fun ok(): String {
    return "OK"
}