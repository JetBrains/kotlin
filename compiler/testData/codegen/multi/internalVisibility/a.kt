package a

val x: String = "OK"