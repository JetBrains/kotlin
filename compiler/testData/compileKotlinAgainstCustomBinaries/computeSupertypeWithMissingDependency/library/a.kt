package a

interface A<T>