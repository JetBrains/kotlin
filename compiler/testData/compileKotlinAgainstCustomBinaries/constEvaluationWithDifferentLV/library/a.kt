package lib

public class Plot {
    public val layout: Int = 0
}

public fun plotFun(plot: Plot): String {
    val mustBeOptimizedByCompiler = plot::layout.name
    return mustBeOptimizedByCompiler
}
