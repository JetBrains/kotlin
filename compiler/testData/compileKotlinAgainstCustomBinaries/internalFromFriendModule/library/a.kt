package a

internal interface InternalInterface

public class PublicClass {
    internal fun internalMemberFun() {}

    internal companion object {}
}

internal val internalVal = ""

internal fun internalFun(s: String): String = s

internal typealias InternalTypealias = InternalInterface
