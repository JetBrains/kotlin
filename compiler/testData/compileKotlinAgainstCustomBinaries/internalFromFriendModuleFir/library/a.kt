package a

internal interface InternalInterface

public class PublicClass internal constructor() {
    internal fun internalMemberFun() {}

    internal companion object {}
}

internal val internalVal = ""

internal fun internalFun(s: String): String = s

internal typealias InternalTypealias = InternalInterface

internal typealias InternalClassAlias = PublicClass