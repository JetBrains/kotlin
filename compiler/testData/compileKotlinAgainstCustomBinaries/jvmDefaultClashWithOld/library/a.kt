package base

interface Check {
    fun test(): String {
        return "fail";
    }

    var test: String
        get() = "123"
        set(value) { value.length}
}

open class CheckClass : Check