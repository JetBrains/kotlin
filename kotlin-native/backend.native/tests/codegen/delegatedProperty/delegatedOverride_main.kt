/*
 * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
 * that can be found in the LICENSE file.
 */

import a.*

open class C: B() {
    override val x: Int = 156

    fun foo() {
        println(x)

        println(super<B>.x)
        bar()
    }
}

fun main(args: Array<String>) {
    val c = C()
    c.foo()
}
