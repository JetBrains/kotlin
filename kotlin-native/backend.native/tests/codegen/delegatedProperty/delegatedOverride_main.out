156
x
117
42
